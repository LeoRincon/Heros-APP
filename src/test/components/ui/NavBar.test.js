import { mount } from 'enzyme';
import { MemoryRouter, Routes, Route } from 'react-router-dom';

import { AuthContext } from '../../../auth/AuthContext';
import { Navbar } from '../../../components/ui/NavBar';
import { types } from '../../../types/index';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('tests in <NavBar /> when is called', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      name: 'Pedro',
      logged: true,
    },
  };
  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter initialEntries={['/login']}>
        <Routes>
          <Route path='/login' element={<Navbar />} />
        </Routes>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test('should be displayed correctly NavBar', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('Pedro');
  });

  test('Should call the logout, call the navigate and dispatch with the arguments', () => {
    wrapper.find('button').simulate('click');

    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.logout,
    });

    expect(mockNavigate).toHaveBeenCalledWith('/login', { replace: true });
  });
});
