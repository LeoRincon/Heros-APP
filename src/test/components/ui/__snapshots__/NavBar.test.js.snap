// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <NavBar /> when is called should be displayed correctly NavBar 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        className="nav-item nav-link "
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav-item nav-link "
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav-item nav-link "
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end px-4"
  >
    <nav
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Pedro
      </span>
      <button
        className="nav-item nav-link btn"
        exact="true"
        onClick={[Function]}
        to="/login"
      >
        Logout
      </button>
    </nav>
  </div>
</nav>
`;
